lightningd/plugin.c

```diff
@@ -256,6 +256,91 @@ static void plugin_request_queue(struct plugin_request *req)
 	io_wake(req->plugin);
 }
 
+static void plugin_log_handle(struct plugin *plugin, const jsmntok_t *paramstok)
+{
+	const jsmntok_t *msgtok, *leveltok;
+	enum log_level level;
+	msgtok = json_get_member(plugin->buffer, paramstok, "message");
+	leveltok = json_get_member(plugin->buffer, paramstok, "level");
+
+	if (!msgtok) {
+		plugin_kill(plugin, "Log notification from plugin doesn't have "
+				    "a \"message\" field");
+		return;
+	}
+
+	if (!leveltok || json_tok_streq(plugin->buffer, leveltok, "info"))
+		level = LOG_INFORM;
+	else if (json_tok_streq(plugin->buffer, leveltok, "debug"))
+		level = LOG_DBG;
+	else if (json_tok_streq(plugin->buffer, leveltok, "warn"))
+		level = LOG_UNUSUAL;
+	else if (json_tok_streq(plugin->buffer, leveltok, "error"))
+		level = LOG_BROKEN;
+	else {
+		plugin_kill(plugin,
+			    "Unknown log-level %.*s, valid values are "
+			    "\"debug\", \"info\", \"warn\", or \"error\".",
+			    json_tok_len(leveltok),
+			    json_tok_contents(plugin->buffer, leveltok));
+		return;
+	}
+
+	/* We strip the \" otherwise they'd be printed here. */
+	log_(plugin->log, level, "log: %.*s", json_tok_len(msgtok) - 2,
```

Comment:
Would be great if there would not only be the message but also the name of the plugin written down to the log. Not sure what's the best way to achieve this. 

I guess having the plugin send their name in the log message. However it would be great to fix the name and map it against registered plugins.

Reply:
`plugin->log` is a logger that is prefixed with the plugin's basename, so the complete line here reads 

```
2018-12-09T12:42:59.072Z plugin-helloworld.py log: Plugin helloworld.py initialized with greeting \"World\"
```

Which I think is what you want.

Reply:
When I Previously used plugin->log I did not see the plugin name. (maybe I was just blind) But maybe I set up my plugin incorrectly or did some other mistake. I will try to see if I can reconstruct this behavior. But it's great that it is already implemented!

Reply:
Was changed recently, in commit 111d6df442d68a7f05fc7c54a2345752ed221cc3

Reply:
I was hoping to differentiate the entries `lightningd` writes about the plugin from the ones that the plugins inject themselves. But you're probably right, it's unlikely that a plugin would inject a log-line pretending to kill itself :-)